src/hotspot/share/opto/predicates.hpp

```diff
@@ -44,9 +45,10 @@ class TemplateAssertionPredicate;
  * - Parse Predicate: Added during parsing to capture the current JVM state. This predicate represents a "placeholder"
  *                    above which Regular Predicates can be created later after parsing.
  *
- *                    There are initially three Parse Predicates for each loop:
+ *                    There are initially four Parse Predicates for each loop:
  *                    - Loop Parse Predicate:             The Parse Predicate added for Loop Predicates.
  *                    - Profiled Loop Parse Predicate:    The Parse Predicate added for Profiled Loop Predicates.
+ *                    - AutoVectorization Predicate:      The Parse Predicate added for AutoVectorization runtime checks.
```

Comment:
Drive-by comment: Can you also add a small section below under "Runtime Predicate" to summarize what an AutoVectorization runtime check is and in what flavors they come? Then we have everything together for future quick reference.